src/script/components/TitleBar/index.tsx

```diff
@@ -0,0 +1,226 @@
+import {ContentViewModel} from '../../view_model/ContentViewModel';
+import {Shortcut} from '../../ui/Shortcut';
+import {ShortcutType} from '../../ui/ShortcutType';
+import cx from 'classnames';
+import {Conversation} from '../../entity/Conversation';
+import {CallingRepository} from '../../calling/CallingRepository';
+import {UserState} from '../../user/UserState';
+import {CallState} from '../../calling/CallState';
+import {TeamState} from '../../team/TeamState';
+import {container} from 'tsyringe';
+import {LegalHoldModalViewModel} from '../../view_model/content/LegalHoldModalViewModel';
+import {ConversationFilter} from '../../conversation/ConversationFilter';
+import {generateWarningBadgeKey} from '../../view_model/content/TitleBarViewModel';
+import {matchQualifiedIds} from 'Util/QualifiedId';
+import {CallActions} from '../../view_model/CallingViewModel';
+import {handleKeyDown} from 'Util/KeyboardUtil';
+
+export interface TitleBarProps {
+  conversationEntity: Conversation;
+  legalHoldModal: LegalHoldModalViewModel;
+  showConversationDetails: (addParticipants: boolean) => void;
+  callingRepository: CallingRepository;
+  callActions: CallActions;
+  isPanelVisible: boolean;
+  userState: UserState;
+  callState: CallState;
+  teamState: TeamState;
+}
+
+const TitleBar: React.FC<TitleBarProps> = ({
+  conversationEntity,
+  legalHoldModal,
+  showConversationDetails,
+  callingRepository,
+  callActions,
+  isPanelVisible,
+  userState = container.resolve(UserState),
+  callState = container.resolve(CallState),
+  teamState = container.resolve(TeamState),
+}) => {
+  const badgeLabelCopy = useMemo(() => {
+    if (conversationEntity.is1to1() && conversationEntity.isRequest()) {
+      return '';
+    }
+
+    const hasExternal = conversationEntity.hasExternal();
+    const hasGuest = conversationEntity.hasDirectGuest();
+    const hasService = conversationEntity.hasService();
+    const hasFederated = conversationEntity.hasFederatedUsers();
+    const translationKey = generateWarningBadgeKey({hasExternal, hasFederated, hasGuest, hasService});
+
+    if (translationKey) {
+      return t(translationKey);
+    }
+
+    return '';
+  }, [conversationEntity]);
+
+  const hasCall = useMemo(() => {
+    const joinedCall = callState.joinedCall();
+    const hasEntities = conversationEntity && !!joinedCall;
+    return hasEntities && matchQualifiedIds(conversationEntity.qualifiedId, joinedCall.conversationId);
+  }, [callState, conversationEntity]);
+
+  const showCallControls = conversationEntity && ConversationFilter.showCallControls(conversationEntity, hasCall);
+
+  const isActivatedAccount = userState.isActivatedAccount();
+
+  const supportsVideoCall = conversationEntity?.supportsVideoCall(callingRepository.supportsConferenceCalling);
+
+  const isVideoCallingEnabled = teamState.isVideoCallingEnabled();
+
+  const conversationSubtitle =
+    conversationEntity.is1to1() && conversationEntity.firstUserEntity?.()?.isFederated
```

Comment:
Those (`is1to1` and `firstUserEntity`) are observables and should be wrapped in a `useKoSubscribableChildren` hook